src/client/src/rt-platforms/platform.ts

```diff
@@ -1,45 +1,64 @@
-const urlParams = new URLSearchParams(window.location.search)
+import { AppConfig, PlatformType } from './types'
+import { Context } from 'openfin-fdc3'
+import { ApplicationEpic } from 'StoreTypes'
+import { Observable } from 'rxjs'
+import { PlatformWindow, PlatformWindowApi } from './platformWindow'
 
-const isFinsemble = 'FSBL' in window
-const isOpenFin = 'fin' in window
-const isGlue42 = 'glue42gd' in window
-const isSymphony = urlParams.has('waitFor') && urlParams.get('waitFor') === 'SYMPHONY'
+export interface PlatformFeatures {
+  app: AppInterop
+  interop: PubSubInterop
+  share: (object: any) => void
+}
 
-export const getSymphonyPlatform = () => import(/* webpackChunkName: "symphony" */ './symphony')
+interface PubSubInterop {
+  subscribe$: (topic: string) => Observable<any>
+  publish: (topic: string, message: any) => void
+}
 
-export const getFinsemblePlatform = () => import(/* webpackChunkName: "finsemble" */ './finsemble')
+interface AppInterop {
+  open: (id: string, config: AppConfig) => Promise<string>
+}
 
-export const getOpenFinPlatform = () => import(/* webpackChunkName: "openfin" */ './openFin')
+export type Platform = Partial<PlatformFeatures> & {
```

Comment:
now since we don't have abstract class anymore, we are using a type which is a combination of optional platform features and other required properties